reset-state

```diff
@@ -60,8 +65,10 @@ set -x
 rm -vrf /snap/*
 set +x
 
-echo
-echo "Starting snapd"
-set -x
-systemctl start snapd
-set +x
+if [ "$snapd_was_active" = "yes" ]; then
+    echo
+    echo "Starting snapd"
+    set -x
+    systemctl start snapd.socket
```

Comment:
note this means that stopping just `snapd` is pointless, as systemd will happily start it again as soon as the socket is accessed. If it is running when you do `systemctl stop snapd` it'll print a warning about this.